### ISSUE TITLE
failing to get private_key from EllipticCurvePrivateNumbers

### ISSUE THREAD

**Issue by simo5** [pyca/cryptography#2397]
Retesting locally an old PR (#1882) to split in pieces I stumble on a few errors:
```
self = <tests.hazmat.primitives.test_ec.TestECDHVectors object at 0x7fee42c6d990>, backend = <cryptography.hazmat.backends.openssl.backend.Backend object at 0x7fee678ef250>
vector = {'CAVS': {'d': 727971953227219448...282703489234652359L, 'x': 573496995671161725...834389382601371856L, 'y': 761249646...69619120228072600L, 'y': 307601656348985140...846650300402439858L}, 'Z': 427597400563304090...422039099186659420L, ...}

    @pytest.mark.parametrize(
        "vector",
        load_vectors_from_file(
            os.path.join(
                "asymmetric", "ECDH",
                "KASValidityTest_ECCStaticUnified_NOKC_ZZOnly_init.fax"),
            load_kasvs_ecdh_vectors
        )
    )
    def test_key_exchange_with_vectors(self, backend, vector):
        curve_type = ec._CURVE_TYPES[vector['curve']]
    
        _skip_curve_unsupported(backend, curve_type)
    
        try:
            key = ec.EllipticCurvePrivateNumbers(
                vector['CAVS']['d'],
                ec.EllipticCurvePublicNumbers(
                    vector['CAVS']['x'],
                    vector['CAVS']['y'],
                    curve_type()
                )
>           ).private_key(backend)

tests/hazmat/primitives/test_ec.py:767: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.tox/py27/lib/python2.7/site-packages/cryptography/hazmat/primitives/asymmetric/ec.py:310: in private_key
    return backend.load_elliptic_curve_private_numbers(self)
.tox/py27/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/backend.py:1652: in load_elliptic_curve_private_numbers
    ec_cdata, public.x, public.y)
.tox/py27/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/backend.py:1793: in _ec_key_set_public_key_affine_coordinates
    self.openssl_assert(res == 0)
.tox/py27/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/backend.py:534: in openssl_assert
    return binding._openssl_assert(self._lib, ok)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

lib = <module 'lib' (built-in)>, ok = False

    def _openssl_assert(lib, ok):
        if not ok:
            errors = _consume_errors(lib)
            raise InternalError(
                "Unknown OpenSSL error. Please file an issue at https://github.com"
                "/pyca/cryptography/issues with information on how to reproduce "
                "this.",
>               errors
            )
E           InternalError: Unknown OpenSSL error. Please file an issue at https://github.com/pyca/cryptography/issues with information on how to reproduce this.

.tox/py27/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py:43: InternalError
```

Does this ring any bell ?

**Comment by alex** [comment]
nope, can you get a pdb in there and see what's in `errors`?

**Comment by simo5** [comment]
Oh interestingly it throws a: ValueError('Invalid EC key.")

**Comment by alex** [comment]
Sorry, what does throws that error, is your original TB not correct?

**Comment by simo5** [comment]
This is failing:
ec.EllipticCurvePublicNumbers(
                    vector['CAVS']['x'],
                    vector['CAVS']['y'],
                    curve_type()
                )

The data comes from the CASV vectors ..

**Comment by simo5** [comment]
So it seem this exception is raised only for sec521r curves where there is a known-fail test where public numbers indeed are wrong and the key is invalid.
But instead of a normal ValueError exception an unown OpenSSL exception is thrown for some reason.

**Comment by simo5** [comment]
errors is empy btw, just pdbed through and the first call to lib.ERR_get_error() just returns 0

**Comment by alex** [comment]
Which values trigger this?

**Comment by simo5** [comment]
Any invalid X or Y value, but only with sec521r, it doesn't happen with the other curves.

**Comment by alex** [comment]
```pytb
>>>> ec.EllipticCurvePrivateNumbers(1, ec.EllipticCurvePublicNumbers(3, 4, ec.SECP521R1())).private_key(default_backend())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alex_gaynor/projects/cryptography/src/cryptography/hazmat/primitives/asymmetric/ec.py", line 289, in private_key
    return backend.load_elliptic_curve_private_numbers(self)
  File "/Users/alex_gaynor/projects/cryptography/src/cryptography/hazmat/backends/multibackend.py", line 253, in load_elliptic_curve_private_numbers
    return b.load_elliptic_curve_private_numbers(numbers)
  File "/Users/alex_gaynor/projects/cryptography/src/cryptography/hazmat/backends/openssl/backend.py", line 1652, in load_elliptic_curve_private_numbers
    ec_cdata, public.x, public.y)
  File "/Users/alex_gaynor/projects/cryptography/src/cryptography/hazmat/backends/openssl/backend.py", line 1789, in _ec_key_set_public_key_affine_coordinates
    raise ValueError("Invalid EC key.")
ValueError: Invalid EC key.
```

**Comment by simo5** [comment]
Try with values from here:
https://github.com/pyca/cryptography/blob/master/vectors/cryptography_vectors/asymmetric/ECDH/KASValidityTest_ECCStaticUnified_NOKC_ZZOnly_init.fax

Look for the values after [EE - SHA512] (line 1598), check the sets marked with results like:
Result = F (2 - CAVS's Static public key Y fails PKV 5.6.2.5)

The public X and Y values are QsCAVSx and QsCAVSy or QsIUTx and QsIUTy (the 2 peers), check which one will fail depending on the "results" description.

I had COUNTs 2,3,5,11,24,25,26,27 raise that internal error IIRC

**Comment by simo5** [comment]
Addressed in #2409
